tests/test_process_response_data.py

```
@@ -102,6 +109,22 @@ def test_process_json_response_with_data_wrapper_without_cast_to(self, client, m
+            {"id": 1, "name": "Test1", "email": "test1@example.com"},
+            {"id": 2, "name": "Test2", "email": "test2@example.com", "admin": True},
+        ]
+        mock_response.json.return_value = test_data
+
+        result = client._process_response_data(mock_response, cast_to=TypeAdapter(list[ResponseTestModel | ResponseTestModelAdmin]))
+
+        assert isinstance(result, list)
+        assert len(result) == 2
+        assert isinstance(result[0], ResponseTestModel) and (result[1], ResponseTestModelAdmin)
+        assert result[0].name == "Test1"
+        assert result[1].admin == True
```

Ruff (0.12.2)

Avoid equality comparisons to `True`; use `result[1].admin:` for truth checks

Replace with `result[1].admin`

(E712)